src/packet.rs

```diff
@@ -467,7 +467,8 @@ pub mod login {
     pub mod clientbound { packets! {
         // 0x00 => Disconnect { reason: Chat }
         0x01 => EncryptionRequest { server_id: String, pubkey: Arr<Var<i32>, u8>, verify_token: Arr<Var<i32>, u8> }
-        // 0x02 => LoginSuccess { uuid: Uuid, username: String; impl Protocol for LoginSuccess { ... } } // NOTE: uuid field is encoded as a string!
+        // NOTE: uuid field is encoded as a string!
+        0x02 => LoginSuccess { uuid: String, username: String }
```

Comment:
Current draft: https://gist.github.com/toqueteos/4549b0b69eef4c1575c6

As atheriel pointed out on IRC, we could impl `FromError<uuid::ParseError>` for `io::Error`.